src/status_im2/contexts/onboarding/create_password/view.cljs

```diff
@@ -187,54 +187,44 @@
-       [rn/view {:style style/flex-fill}
-        [rn/keyboard-avoiding-view {:style style/flex-fill}
-         [navigation-bar/navigation-bar
-          {:top                   top
-           :right-section-buttons [{:type                :blur-bg
-                                    :icon                :i/info
-                                    :icon-override-theme :dark
-                                    :on-press            (fn []
-                                                           (rn/dismiss-keyboard!)
-                                                           (rf/dispatch [:show-bottom-sheet
-                                                                         {:content create-password-doc
-                                                                          :shell?  true}]))}]}]
-         [password-form]
-         [rn/view {:style {:height (if-not @keyboard-shown? bottom 0)}}]]]])))
+    [quo/text {:size :paragraph-2}
+     (i18n/label :t/create-profile-password-info-box-description)]]])
 
 (defn create-password
   []
-  [:<>
-   [background/view true]
-   [:f> f-create-password]])
+  (reagent/with-let [keyboard-shown?      (reagent/atom false)
+                     {:keys [top bottom]} (safe-area/get-insets)
+                     will-show-listener   (ocall rn/keyboard
+                                                 "addListener"
+                                                 "keyboardWillShow"
+                                                 #(reset! keyboard-shown? true))
+                     will-hide-listener   (ocall rn/keyboard
+                                                 "addListener"
+                                                 "keyboardWillHide"
+                                                 #(reset! keyboard-shown? false))
+                     on-press-info        (fn []
+                                            (rn/dismiss-keyboard!)
+                                            (rf/dispatch [:show-bottom-sheet
+                                                          {:content create-password-doc
+                                                           :shell?  true}]))]
+    [:<>
+     [background/view true]
+     [rn/touchable-without-feedback
+      {:on-press   rn/dismiss-keyboard!
+       :accessible false}
+      [rn/view {:style style/flex-fill}
+       [rn/keyboard-avoiding-view {:style style/flex-fill}
+        [navigation-bar/navigation-bar
+         {:top                   top
+          :right-section-buttons [{:type                :blur-bg
+                                   :icon                :i/info
+                                   :icon-override-theme :dark
+                                   :on-press            on-press-info}]}]
+        [password-form]
+        [rn/view {:style {:height (if-not @keyboard-shown? bottom 0)}}]]]]]
+    (finally
+     (ocall will-show-listener "remove")
+     (ocall will-hide-listener "remove"))))
```

Comment:
Ok, I added this as a separate commit in case you think it should be rolled back. I refactored the pattern we had to avoid using a functional component and the `use-effect` hook. IMO, it is more clear, wdyt?

To be clear, we had:

```clojure
(defn- f-my-component []
  (let [initialization vars]
    (fn []
      (rn/use-effect
       ;; some initialization code
       (fn []
         ;; some cleaning code
         ))
      [rn/something
       ;; View code
       ])))

(defn my-component []
  [:f> f-my-component])
  ```
  
  and now we have:
  ```clojure
  (defn my-component []
  (reagent/with-let [initialization vars
                     ;; some initialization code
                     ]
    [rn/something
     ;;View code
     ]
    (finally
     ;; some cleaning code
     )))
  ```
  
  What I think is better:
-   There's no need to use a functional component to remove the listeners on unmount,
-  The listeners refs are available for all the function, not only inside the use-effect fn
- We don't need to wrap the component and use the `:f>` keyword

Reply:
I couldn't agree more @ulisesmac. It's cleaner.

On a separate note, not for this PR. I noticed that a quite number of the screens in our codebase need to know when the keyboard is open. I thought of having a global keyboard listener where we can update the re-frame app-db and the screens/components which need it can subscribe to it. It would reduce a lot of keyboard listener code. WDYT?

Reply:
@smohamedjavid Sounds good! IDK if we can add those listeners during app initialization, but if so, it'd be a very nice improvement.